datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -952,7 +1031,15 @@ private void initializeTask(DatastreamTask task, Set<DatastreamTask> failedDatas
       boolean customCheckpointing = getCustomCheckpointing(task);
       TransportProviderAdmin tpAdmin = _transportProviderAdmins.get(task.getTransportProviderName());
       TransportProvider transportProvider = tpAdmin.assignTransportProvider(task);
-      EventProducer producer = new EventProducer(task, transportProvider, _cpProvider, _eventProducerConfig, customCheckpointing);
+
+      Function<DatastreamTask, Set<String>> throughputViolatingTopicsProvider =
+          _config.getEnableThroughputViolatingTopicsHandling() ?
+              (t) -> getThroughputViolatingTopics(t.getDatastreams()) :
+              (t) -> new HashSet<>();
+
+      EventProducer producer =
```

Comment:
From @vmaheshw 
```
Will it make sense to simplify this at Datastream level rather than DatastreamTask level?
For eg: in digest, there is only one datastream. This function will result in 10 copies for 
the same datastream.

Also, hashing based on datastream name will be cheaper and more efficient.
```

Reply:
Instead of moving this at the Datastream level, I refactored to only maintaining a single initialization of this callback per instance and only maintaining the set of violations per init of EventProducer.